simpleio.py

```diff
@@ -32,8 +32,96 @@
 
 import digitalio
 import math
+import time
 from neopixel_write import neopixel_write
 
+
+def tone(pin, frequency, duration=0):
+    """
+    Generates a square wave of the specified frequency (50% duty cycle)
+    on a pin
+
+    :param pin: the pin on which to generate the tone
+    :param frequency: frequency of tone in Hz
+    :param duration: duration of tone in millis (optional)
+    """
+    for i in range(0, duration):
+        pwm = pulseio.PWMOut(pin, duty_cycle = 2 ** 15, frequency, variable_frequency = False)
+        time.sleep(0.001)
+
+
+def noTone(pin):
+    """
+    Stops the generation of the square wave produced by tone()
+    by deinit'ing the pin.
+
+    :param pin: pin on which to stop the tone from being played
+    """
+    pwm.deinit(pin)
```

Comment:
Have you tried this? I don't think the pwm object will persist across the `tone` and `noTone` calls. It also doesn't take a pin in.

Adding support for the analog out pin would also be nice. PWM doesn't work with it but AudioOut does.